ambry-store/src/main/java/com.github.ambry.store/IndexValue.java

```diff
@@ -13,114 +13,329 @@
  */
 package com.github.ambry.store;
 
+import com.github.ambry.utils.Time;
 import com.github.ambry.utils.Utils;
 import java.nio.ByteBuffer;
 
 
 /**
  * Represents the blob value stored in the index for a key.
+ *
+ * Version_0
+ * - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+ * | Blob Size |   Offset  |  Flags  | Expiration Time | Orig msg  |
+ * | (8 bytes) | (8 bytes) | (1 byte)|     in Ms       | offset    |
+ * |           |           |         |     ( 8 bytes)  | (8 bytes) |
+ * - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+ *
+ * Version_1
+ * - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+ * | Blob Size |   Offset  |  Flags  | Expiration Time | Orig msg  | OperationTime | ServiceId | ContainerId |
+ * | (8 bytes) | (8 bytes) | (1 byte)|   in  Secs      | offset    |  (4 bytes)    | (2 bytes) | (2 bytes)   |
+ * |           |           |         |     ( 4 bytes)  | (8 bytes) |               |           |             |
+ * - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+ *
  */
-
 class IndexValue {
   enum Flags {
     Delete_Index
   }
 
-  private static int Blob_Size_In_Bytes = 8;
-  private static int Offset_Size_In_Bytes = 8;
-  private static int Flag_Size_In_Bytes = 1;
-  private static int Expires_At_Ms_Size_In_Bytes = 8;
-  private static int Original_Message_Offset_Size_In_Bytes = 8;
+  private final static int BLOB_SIZE_IN_BYTES = 8;
+  private final static int OFFSET_SIZE_IN_BYTES = 8;
+  private final static int FLAG_SIZE_IN_BYTES = 1;
+  private final static int EXPIRES_AT_MS_SIZE_IN_BYTES_V0 = 8;
+  private final static int ORIGINAL_MESSAGE_OFFSET_SIZE_IN_BYTES = 8;
+
+  private final static int EXPIRES_AT_SECS_SIZE_IN_BYTES_V1 = 4;
+  private final static int OPERATION_TIME_SECS_SIZE_IN_BYTES = 4;
+  private final static int SERVICE_ID_SIZE_IN_BYTES = 2;
+  private final static int CONTAINER_ID_SIZE_IN_BYTES = 2;
 
-  static int Index_Value_Size_In_Bytes =
-      Blob_Size_In_Bytes + Offset_Size_In_Bytes + Flag_Size_In_Bytes + Expires_At_Ms_Size_In_Bytes
-          + Original_Message_Offset_Size_In_Bytes;
+  private final int indexValueSizeInBytes;
 
-  private final ByteBuffer value;
+  private long size;
   private Offset offset;
+  private byte flags;
+  private final long expiresAtMs;
+  private long originalMessageOffset;
```

Comment:
Original offset will be set only if the put record belongs to the same log segment as delete record. Hence Offset object is not required, just the long offset is good enough